rep_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The go-zeromq Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package zmq4_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-zeromq/zmq4"
+)
+
+func TestIssue99(t *testing.T) {
+	var (
+		wg     sync.WaitGroup
+		outMsg zmq4.Msg
+		inMsg  zmq4.Msg
+	)
+
+	ep, err := EndPoint("tcp")
+	if err != nil {
+		t.Errorf("could not find endpoint: %v", err)
+	}
+
+	requester := func() {
+
+		req := zmq4.NewReq(context.Background())
+		defer req.Close()
+		defer wg.Done()
+
+		err := req.Dial(ep)
+		if err != nil {
+			t.Errorf("could not dial: %v", err)
+		}
+
+		// Test message w/ 3 frames
+		outMsg = zmq4.NewMsgFromString([]string{"ZERO", "Hello!", "World!"})
+		err = req.Send(outMsg)
+		if err != nil {
+			t.Errorf("failed to send: %v", err)
+		}
+
+		inMsg, err = req.Recv()
+		if err != nil {
+			t.Errorf("failed to recv: %v", err)
+		}
+	}
+
+	responder := func() {
+
+		rep := zmq4.NewRep(context.Background())
+		defer rep.Close()
+		defer wg.Done()
```

Comment:
shouldn't this `defer` be the first (so the last to execute)?

Reply:
I don't think does matter in this case -- it's a test.
The sockets Close() will be called which frees the resources.

Reply:
sure, it's a test, but in my experience, I noticed a lot of code is written by copy-pasting.
in effect, bad code can spread like wild fire "because it was just to show something" or "it was just a test".

I certainly don't claim I am clear of never having cut that corner, but I try not to.
could you please move that `defer` up?
thanks a lot.